File: minecraft/fix_fluids.py
import os

scripts_dir = 'c:/Users/wamb9/MINECRAFT/Prism/Instances/PEAK dev/minecraft/kubejs/server_scripts/'

# Fix 00_recipe_error_suppressor.js
path = os.path.join(scripts_dir, '00_recipe_error_suppressor.js')
with open(path, 'r', encoding='utf-8') as f:
    text = f.read()
text = text.replace("Fluid.of('#c:crude_oil'", "Fluid.of('createdieselgenerators:crude_oil'")
text = text.replace("Fluid.of('#c:plantoil'", "Fluid.of('createdieselgenerators:plant_oil'")
text = text.replace("Fluid.of('#c:ethanol'", "Fluid.of('createdieselgenerators:ethanol'")
with open(path, 'w', encoding='utf-8') as f:
    f.write(text)

# Fix 20_fixed_datapacks.js
path = os.path.join(scripts_dir, '20_fixed_datapacks.js')
with open(path, 'r', encoding='utf-8') as f:
    text = f.read()
text = text.replace("Fluid.of('c:molten_netherite'", "Fluid.of('createmetalwork:molten_netherite'")
with open(path, 'w', encoding='utf-8') as f:
    f.write(text)
print('Fixed fluids!')

File: fixFluids.js
const fs = require('fs');
const path = 'C:/Users/chris/AppData/Roaming/PrismLauncher/instances/PEAK-dev/minecraft/kubejs/server_scripts/00_recipe_error_suppressor.js';
let content = fs.readFileSync(path, 'utf8');

// For fluids in parseIng
content = content.replace(/if \(i\.fluid\) return \{ fluid: i\.fluid, amount: i\.amount \};/g, "if (i.fluid) return { type: 'neoforge:single', fluid: i.fluid, amount: i.amount };");

// For fluid tags in parseIng
content = content.replace(/if \(i\.fluidTag\) return \{ fluidTag: i\.fluidTag, amount: i\.amount \};/g, "if (i.fluidTag) return { type: 'neoforge:tag', tag: i.fluidTag, amount: i.amount };");

// Add cement to fluid list
content = content.replace(/includes\('venom'\)/g, "includes('venom') || id.includes('cement')");

fs.writeFileSync(path, content, 'utf8');
console.log('Fixed fluid ingredients in ' + path);
